server/server.go

```diff
@@ -1413,7 +1413,7 @@ func (server *ArgoCDServer) getIndexData() ([]byte, error) {
 }
 
 func (server *ArgoCDServer) uiAssetExists(filename string) bool {
-	f, err := server.staticAssets.Open(strings.Trim(filename, "/"))
+	f, err := server.staticAssets.Open(filepath.Clean(filename))
```

Comment:
`server.staticAssets` by default only serves an embedded FS, so it shouldn't be vulnerable to traversal.

If the user has specified a `--staticassets` dir on the API deployment, they expand `server.staticAssets` to serve on-disk files. By my understanding of `os.DirFS` can only be escaped via symlinks, not just paths. Either way I'd want to see a PoC of a working exploit.

I think a better way to harden the `--staticassets` feature would be to use `root.FS` instead of `os.DirFS`.

Reply:
https://github.com/argoproj/argo-cd/pull/22936

Reply:
Thanks, I guess I can close this PR then.

Reply:
Sounds good! Let me know if you have feedback on my PR's approach